src/utils/tests/test_strutils.c

```diff
@@ -187,6 +187,80 @@ MU_TEST(test_strhex_decode)
 }
 
 
+MU_TEST(test_strcategory)
+{
+  mu_assert_int_eq(strcatUpper, strcategory('A'));
+  mu_assert_int_eq(strcatUpper, strcategory('Z'));
+  mu_assert_int_eq(strcatLower, strcategory('a'));
+  mu_assert_int_eq(strcatLower, strcategory('z'));
+  mu_assert_int_eq(strcatDigit, strcategory('0'));
+  mu_assert_int_eq(strcatDigit, strcategory('9'));
+  mu_assert_int_eq(strcatUnreserved, strcategory('-'));
+  mu_assert_int_eq(strcatUnreserved, strcategory('.'));
+  mu_assert_int_eq(strcatUnreserved, strcategory('_'));
+  mu_assert_int_eq(strcatUnreserved, strcategory('~'));
+  mu_assert_int_eq(strcatReserved, strcategory(':'));
+  mu_assert_int_eq(strcatReserved, strcategory('/'));
+  mu_assert_int_eq(strcatReserved, strcategory('?'));
+  mu_assert_int_eq(strcatReserved, strcategory('#'));
+  mu_assert_int_eq(strcatReserved, strcategory('['));
+  mu_assert_int_eq(strcatReserved, strcategory(']'));
+  mu_assert_int_eq(strcatReserved, strcategory('@'));
+  mu_assert_int_eq(strcatSpecific, strcategory('!'));
+  mu_assert_int_eq(strcatSpecific, strcategory('$'));
+  mu_assert_int_eq(strcatSpecific, strcategory('&'));
+  mu_assert_int_eq(strcatSpecific, strcategory('\''));
+  mu_assert_int_eq(strcatSpecific, strcategory('('));
+  mu_assert_int_eq(strcatSpecific, strcategory(')'));
+  mu_assert_int_eq(strcatSpecific, strcategory('*'));
+  mu_assert_int_eq(strcatSpecific, strcategory('+'));
+  mu_assert_int_eq(strcatSpecific, strcategory(','));
+  mu_assert_int_eq(strcatSpecific, strcategory(';'));
+  mu_assert_int_eq(strcatSpecific, strcategory('='));
+  mu_assert_int_eq(strcatPercent, strcategory('%'));
+  mu_assert_int_eq(strcatNul, strcategory('\0'));
+  mu_assert_int_eq(strcatOther, strcategory('"'));
+  mu_assert_int_eq(strcatOther, strcategory('`'));
+  mu_assert_int_eq(strcatOther, strcategory('<'));
+  mu_assert_int_eq(strcatOther, strcategory('>'));
```

Comment:
What happens if the character is non-7-bit ascii?

Reply:
It should be an other. Added a test to confirm it :-)